src/gltfPreview.ts

```diff
@@ -1,39 +1,26 @@
-'use strict';
 import * as vscode from 'vscode';
 import * as path from 'path';
 import * as fs from 'fs';
-import { ExtensionContext, TextDocumentContentProvider, EventEmitter, Event, Uri, ViewColumn } from 'vscode';
+import { ContextBase } from './contextBase';
+import { toResourceUrl, parseJsonMap } from './utilities';
 
-export class GltfPreviewDocumentContentProvider implements TextDocumentContentProvider {
-    private _onDidChange = new EventEmitter<Uri>();
-    private _context: ExtensionContext;
-    private _mainHtml: string;
-    private _babylonHtml: string;
-    private _cesiumHtml: string;
-    private _threeHtml: string;
+export class GltfPreview extends ContextBase {
+    private readonly _mainHtml: string;
+    private readonly _babylonHtml: string;
+    private readonly _cesiumHtml: string;
+    private readonly _threeHtml: string;
 
-    public UriPrefix = 'gltf-preview://';
+    private _panels: { [fileName: string]: vscode.WebviewPanel } = {};
 
-    constructor(context: ExtensionContext) {
-        this._context = context;
-        this._mainHtml = fs.readFileSync(this._context.asAbsolutePath('pages/previewModel.html'), 'UTF-8')
+    constructor(context: vscode.ExtensionContext) {
+        super(context);
+
+        this._mainHtml = fs.readFileSync(this._context.asAbsolutePath('pages/previewModel.html'), 'UTF-8');
         this._babylonHtml = encodeURI(fs.readFileSync(this._context.asAbsolutePath('pages/babylonView.html'), 'UTF-8'));
         this._cesiumHtml = encodeURI(fs.readFileSync(this._context.asAbsolutePath('pages/cesiumView.html'), 'UTF-8'));
         this._threeHtml = encodeURI(fs.readFileSync(this._context.asAbsolutePath('pages/threeView.html'), 'UTF-8'));
     }
 
-    private addFilePrefix(file: string): string {
-        return ((file[0] === '/') ? 'file://' : 'file:///') + file;
```

Comment:
Has this branch been tested on MacOS and Linux?  This stuff was added in #73 to support those platforms, it's not needed on Windows.

Reply:
I haven't, no. It shouldn't be an issue anymore though since webviews must use vscode-resource prefix to access resources. I'll try it on a Mac to be sure.

Reply:
No I think Mac and Linux are similar enough in the file handling that Linux should be OK too.  Thanks for testing on Mac.